### PR TITLE
refactor(melange): switch `--bs-package-name` to library name

### PR DESCRIPTION
**Pull request by anmonteiro**
Signed-off-by: Antonio Nuno Monteiro <anmonteiro@gmail.com>

### CODE REVIEW

File: src/dune_rules/compilation_context.mli [hunk 1]
```diff
@@ -85,6 +86,8 @@ val package : t -> Package.t option
 
 val vimpl : t -> Vimpl.t option
 
+val lib_name : t -> Lib_name.t option
```
Comment (rgrinberg) [inline src/dune_rules/compilation_context.mli:89]: How about changing the name to reflect that this is only for public libraries? `public_lib_name` for example.

Comment (anmonteiro) [reply src/dune_rules/compilation_context.mli]: Good idea, changed it in bf43e9b

**Review by rgrinberg — APPROVED** [review]
The change looks correct to me. Let's hear from @jchavarri before merging.

File: src/dune_rules/lib_rules.ml [hunk 2]
```diff
@@ -457,10 +458,19 @@ let cctx (lib : Library.t) ~sctx ~source_modules ~dir ~expander ~scope
   let js_of_ocaml =
     Js_of_ocaml.In_context.make ~dir lib.buildable.js_of_ocaml
   in
+  (* XXX(anmonteiro): `public_lib_name` is used to derive Melange's
```
Comment (jchavarri) [inline src/dune_rules/lib_rules.ml:461]: Just curious: what does XXX stand for, or what is the meaning? I generally thought it's used on temporary changes (to replace FIXME, TODO and such). But in this case, the change is most probably permanent right?

**Review by jchavarri — APPROVED** [review]
Looks great. Tested with Ahrefs codebase, no issues found.

File: src/dune_rules/lib_rules.ml (hunk 2 above)
Comment (anmonteiro) [reply src/dune_rules/lib_rules.ml:461]: I generally use it in the way this answer explains: https://stackoverflow.com/questions/1452934/what-is-the-meaning-of-xxx-in-code-comments

Perhaps a `NOTE:` would be more appropriate in this case.

**Pull request merged**